tardis/plugins/sqliteregistry.py

```diff
@@ -33,25 +34,48 @@ def __init__(self):
             for machine_type in getattr(configuration, site.name).MachineTypes:
                 self.add_machine_types(site.name, machine_type)
 
-    def add_machine_types(self, site_name: str, machine_type: str):
+    def add_machine_types(self, site_name: str, machine_type: str) -> None:
+        if self.get_machine_type(site_name, machine_type):
+            logger.debug(
+                f"{machine_type} is already present for {site_name} in database! Skipping insertion!"  # noqa B950
+            )
+            return
         sql_query = """
-        INSERT OR IGNORE INTO MachineTypes(machine_type, site_id)
+        INSERT OR ROLLBACK INTO MachineTypes(machine_type, site_id)
         SELECT :machine_type, Sites.site_id FROM Sites
         WHERE Sites.site_name = :site_name"""
         self.execute(sql_query, dict(site_name=site_name, machine_type=machine_type))
 
-    def add_site(self, site_name: str):
-        sql_query = "INSERT OR IGNORE INTO Sites(site_name) VALUES (:site_name)"
+    def get_machine_type(self, site_name: str, machine_type: str) -> List[Dict]:
+        sql_query = """
+        SELECT * FROM MachineTypes MT
+        JOIN Sites S ON MT.site_id = S.site_id
+        WHERE MT.machine_type = :machine_type AND S.site_name = :site_name"""
+        return self.execute(
+            sql_query, dict(site_name=site_name, machine_type=machine_type)
+        )
+
+    def add_site(self, site_name: str) -> None:
+        if self.get_site(site_name):
+            logger.debug(
+                f"{site_name} already present in database! Skipping insertion!"
+            )
+            return
+        sql_query = "INSERT OR ROLLBACK INTO Sites(site_name) VALUES (:site_name)"
         self.execute(sql_query, dict(site_name=site_name))
 
-    async def async_execute(self, sql_query: str, bind_parameters: dict):
+    def get_site(self, site_name: str) -> List[Dict]:
+        sql_query = "SELECT * FROM Sites WHERE site_name = :site_name"
+        return self.execute(sql_query, dict(site_name=site_name))
+
+    async def async_execute(self, sql_query: str, bind_parameters: dict) -> List[Dict]:
         loop = asyncio.get_event_loop()
         return await loop.run_in_executor(
             self.thread_pool_executor, self.execute, sql_query, bind_parameters
         )
 
     @contextmanager
-    def connect(self):
+    def connect(self) -> None:
```

Comment:
I think this should be ``-> Generator[sqlite3.Connection, None, None]`` instead; ``-> Iterator[sqlite3.Connection]`` might work as well. The `@contextmanager` is annotated so that a type checker should understand what is going on. (It just [won't work properly for PyCharm](https://youtrack.jetbrains.com/issue/PY-36444), though.)